feature-libs/checkout/base/core/facade/checkout-delivery-modes.service.ts

```diff
@@ -0,0 +1,240 @@
+      () =>
+        this.checkoutPreconditions().pipe(
+          switchMap(([userId, cartId]) =>
+            this.checkoutDeliveryModesConnector.getSupportedModes(
+              userId,
+              cartId
+            )
+          )
+        ),
+      {
+        reloadOn: this.getSupportedDeliveryModesReloadTriggers(),
+        resetOn: this.getSupportedDeliveryModesQueryResetTriggers(),
+        retryOn: { shouldRetry: isJaloError },
+      }
+    );
+
+  protected setDeliveryModeCommand: Command<string, unknown> =
+    this.command.create<string>(
+      (deliveryModeCode) =>
+        this.checkoutPreconditions().pipe(
+          switchMap(([userId, cartId]) => {
+            if (!deliveryModeCode) {
+              throw new Error('Checkout conditions not met');
+            }
+
+            return this.checkoutDeliveryModesConnector
+              .setMode(userId, cartId, deliveryModeCode)
+              .pipe(
+                tap(() => {
+                  this.eventService.dispatch(
+                    {
+                      userId,
+                      cartId,
+                      deliveryModeCode,
+                    },
+                    DeliveryModeSetEvent
+                  );
+                  /**
+                   * TODO: We have to keep this here, since the cart feature is still ngrx-based.
+                   * Remove once it is switched from ngrx to c&q.
+                   * We should dispatch an event, which will load the cart$ query.
+                   */
+                  this.store.dispatch(
+                    new CartActions.LoadCart({
+                      userId,
+                      cartId,
+                    })
+                  );
+                })
+              );
+          })
+        ),
+      {
+        strategy: CommandStrategy.CancelPrevious,
+      }
+    );
+
+  protected clearDeliveryModeCommand: Command<void, unknown> =
+    this.command.create<void>(
+      () =>
+        this.checkoutPreconditions().pipe(
+          switchMap(([userId, cartId]) =>
+            this.checkoutDeliveryModesConnector
+              .clearCheckoutDeliveryMode(userId, cartId)
+              .pipe(
+                tap(() => {
+                  this.eventService.dispatch(
+                    {
+                      userId,
+                      cartId,
+                    },
+                    DeliveryModeClearedEvent
+                  );
+                  /**
+                   * TODO: We have to keep this here, since the cart feature is still ngrx-based.
+                   * Remove once it is switched from ngrx to c&q.
+                   * We should dispatch an event, which will reload the cart$ query.
+                   */
+                  this.store.dispatch(
+                    new CartActions.LoadCart({
+                      cartId,
+                      userId,
+                    })
+                  );
+                }),
+                catchError((error) => {
```

Comment:
can you remind me this

I understand that's what we did for the previous ngrx way of checkout, but we caught the error in the occ..adapter too, so the thrown error there is being caught here as well right since it bubbles up

This would be like a special case, where the we have to catch the error here too to do something else as we want to keep all service layer code in here instead of dispatching it in the adapter right?

Reply:
Yes, this is a special cases which was added to mimic the old behavior where we are reloading the cart on error.